### PR TITLE
Add a few miscellaneous pieces of documentation

### PR DESCRIPTION
**Pull request by ajreynol**

### CODE REVIEW

File: docs/theories/datatypes.rst [hunk 1]
```diff
@@ -109,6 +111,27 @@ Examples
...
+
+Datatype updaters are a (non-standard) extension available in datatype logics.
+The term:
+
+.. code:: smtlib
+   ((_ update Sij) t u)
+
+is equivalent to replacing the field of ``t`` denoted by the selector ``Sij``
+with the value ``u``, or ``t`` itself if that selector does not apply to the
+constructor symbol of ``t``.  For example, for the list datatype, we have that:
```
Comment (aniemetz) [inline docs/theories/datatypes.rst:125]: Isn't it easier to say that it doesn't get updated if the selector does not apply?

**Review by aniemetz — COMMENTED** [review]
Can you add the changes in cvc5.h to Python and Java, too?

File: docs/theories/datatypes.rst (hunk 1 above)
Comment (ajreynol) [reply docs/theories/datatypes.rst:125]: To me, this would suggest that terms are stateful.  Its better to say that the updater returns `t`.

**Comment by ajreynol** [comment]
The Java API actually does not have comments above any class, @mudathirmahgoub can you confirm?

Should we copy the documentation above the C++ classes to all Java classes?

**Comment by mudathirmahgoub** [comment]
> The Java API actually does not have comments above any class, @mudathirmahgoub can you confirm?
> 
> Should we copy the documentation above the C++ classes to all Java classes?
Yes. That should work .

File: src/api/java/io/github/cvc5/DatatypeDecl.java [hunk 2]
```diff
@@ -15,6 +15,18 @@
 
 package io.github.cvc5;
 
+/**
+ * A cvc5 datatype declaration. A datatype declaration is not itself a datatype
+ * (see Datatype), but a specification for creating a datatype sort.
```
Comment (mudathirmahgoub) [inline src/api/java/io/github/cvc5/DatatypeDecl.java:20]:
```suggestion
 * (see {@link Datatype}), but a specification for creating a datatype sort.
```

File: src/api/java/io/github/cvc5/DatatypeDecl.java [hunk 3]
```diff
@@ -15,6 +15,18 @@
...
+ *
+ * The interface for a datatype declaration coincides with the syntax for the
+ * SMT-LIB 2.6 command `declare-datatype`, or a single datatype within the
+ * `declare-datatypes` command.
+ *
+ * Datatype sorts can be constructed from DatatypeDecl using the methods:
+ *   - Solver::mkDatatypeSort()
```
Comment (mudathirmahgoub) [inline src/api/java/io/github/cvc5/DatatypeDecl.java:27]:
```suggestion
 *   - {@link Solver#mkDatatypeSort(DatatypeDecl)}
```

File: src/api/java/io/github/cvc5/DatatypeDecl.java [hunk 4]
```diff
@@ -15,6 +15,18 @@
...
+ *   - Solver::mkDatatypeSorts()
```
Comment (mudathirmahgoub) [inline src/api/java/io/github/cvc5/DatatypeDecl.java:28]:
```suggestion
 *   - {@link Solver#mkDatatypeSorts(DatatypeDecl[])}
```

File: src/api/java/io/github/cvc5/Op.java [hunk 5]
```diff
@@ -15,6 +15,11 @@
 
 package io.github.cvc5;
 
+/**
+ * A cvc5 operator.
+ * An operator is a term that represents certain operators, instantiated
+ * with its required parameters, e.g., a term of kind BITVECTOR_EXTRACT.
```
Comment (mudathirmahgoub) [inline src/api/java/io/github/cvc5/Op.java:21]:
```suggestion
 * with its required parameters, e.g., a term of kind {@link Kind#BITVECTOR_EXTRACT}.
```

File: src/api/java/io/github/cvc5/SynthResult.java [hunk 6]
```diff
@@ -15,6 +15,16 @@
 
 package io.github.cvc5;
 
+/**
+ * Encapsulation of a solver synth result.
+ *
+ * This is the return value of the API methods:
+ *   - Solver::checkSynth()
```
Comment (mudathirmahgoub) [inline src/api/java/io/github/cvc5/SynthResult.java:22]:
```suggestion
 *   - {@link Solver#checkSynth()}
```

File: src/api/java/io/github/cvc5/SynthResult.java [hunk 7]
```diff
@@ -15,6 +15,16 @@
...
+ *   - Solver::checkSynthNext()
```
Comment (mudathirmahgoub) [inline src/api/java/io/github/cvc5/SynthResult.java:23]:
```suggestion
 *   - {@link Solver#checkSynthNext()}
```

File: src/api/python/cvc5.pxi [hunk 8]
```diff
@@ -397,7 +398,17 @@ cdef class DatatypeConstructorDecl:
 
 cdef class DatatypeDecl:
     """
-        A cvc5 datatype declaration.
+        A cvc5 datatype declaration. A datatype declaration is not itself a
+        datatype (see Datatype), but a specification for creating a datatype
```
Comment (aniemetz) [inline src/api/python/cvc5.pxi:402]:
```suggestion
        datatype (see :py:class:`cvc5.Datatype`), but a specification for creating a datatype
```

File: src/api/python/cvc5.pxi [hunk 9]
```diff
@@ -397,7 +398,17 @@ cdef class DatatypeConstructorDecl:
...
+        sort.
+
+        The interface for a datatype declaration coincides with the syntax for
+        the SMT-LIB 2.6 command `declare-datatype`, or a single datatype within
+        the `declare-datatypes` command.
+
+        Datatype sorts can be constructed from DatatypeDecl using the methods:
```
Comment (aniemetz) [inline src/api/python/cvc5.pxi:409]:
```suggestion
        Datatype sorts can be constructed from :py:class:`DatatypeDecl` using
        the methods:
        
```

File: src/api/python/cvc5.pxi [hunk 10]
```diff
@@ -397,7 +398,19 @@ cdef class DatatypeConstructorDecl:
...
+
+        The interface for a datatype declaration coincides with the syntax for
+        the SMT-LIB 2.6 command `declare-datatype`, or a single datatype within
+        the `declare-datatypes` command.
+
+        Datatype sorts can be constructed from :py:class:`DatatypeDecl` using
+        the methods:
+        
+            - :py:meth:`Solver::mkDatatypeSort`
+            - :py:meth:`Solver::mkDatatypeSorts`
```
Comment (aniemetz) [inline src/api/python/cvc5.pxi:413]:
```suggestion
            - :py:meth:`Solver.mkDatatypeSort()`
            - :py:meth:`Solver.mkDatatypeSorts()`
```

**Review by aniemetz — APPROVED** [review]

**Pull request merged**